scripts/Test-PendingReboot.ps1

```diff
@@ -0,0 +1,23 @@
+<#
+    Return 0 / SUCCESS if reboot is required!
+#>
+
+$cbsKey = Get-ItemProperty "HKLM:\Software\Microsoft\Windows\CurrentVersion\Component Based Servicing" -ErrorAction Ignore
+if ($cbsKey.PSObject.Properties.name -contains 'RebootPending') {
+    Write-Output "Reboot requried - Component Based Servicing:RebootPending registry value detected"
+    New-Item -Path '/var/run' -Name 'reboot-required' -ItemType File -Force
+}
+
+$autoUpdateKey = Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update" -ErrorAction Ignore
+if ($autoUpdateKey.PSObject.Properties.name -contains 'RebootRequired') {
+    Write-Output "Reboot required - Auto Update:RebootRequired registry value detected"
```

Comment:
Write-Output returns objects to the pipeline. We do not use it in a script unless that behavior is expected. Instead, you can use Write-Information.

PS C:\scripts> .\Test-PendingReboot.ps1 
Reboot not required